pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -106,27 +106,50 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: buildRun.Spec.BuildRef.Name, Namespace: buildRun.Namespace}, build)
 	if err != nil {
 		reqLogger.Error(err, "Failed to get Build from BuildRun", "Build", buildRun.Spec.BuildRef.Name)
+		buildRun = r.updateBuildRunErrorStatus(buildRun, err)
+		err = r.client.Status().Update(context.TODO(), buildRun)
+		if err != nil {
+			reqLogger.Error(err, "Failed to update the BuildRun status", "BuildRun", buildRun.Name)
+			return reconcile.Result{}, err
+		}
 		return reconcile.Result{}, err
 	}
 
 	lastTaskRun, err := r.retrieveTaskRun(build, buildRun)
 	if err != nil {
 		reqLogger.Error(err, "Failed to list existing TaskRuns from BuildRun", "BuildRun", buildRun.Name)
+		buildRun = r.updateBuildRunErrorStatus(buildRun, err)
+		err = r.client.Status().Update(context.TODO(), buildRun)
+		if err != nil {
+			reqLogger.Error(err, "Failed to update the BuildRun status", "BuildRun", buildRun.Name)
+			return reconcile.Result{}, err
+		}
 		return reconcile.Result{}, err
 	}
 
 	if lastTaskRun != nil {
 		// TODO: Make this safer
 		if len(lastTaskRun.Status.Conditions) > 0 {
-			buildRun.Status.Succeeded = lastTaskRun.Status.Conditions[0].Status
-			buildRun.Status.Reason = lastTaskRun.Status.Conditions[0].Reason
+			taskRunStatus := lastTaskRun.Status.Conditions[0].Status
+			buildRun.Status.Succeeded = taskRunStatus
+			if taskRunStatus == corev1.ConditionFalse {
+				buildRun.Status.Reason = lastTaskRun.Status.Conditions[0].Message
+			} else {
+				buildRun.Status.Reason = lastTaskRun.Status.Conditions[0].Reason
+			}
 		}
 		buildRun.Status.LatestTaskRunRef = &lastTaskRun.Name
 		buildRun.Status.StartTime = lastTaskRun.Status.StartTime
 		buildRun.Status.CompletionTime = lastTaskRun.Status.CompletionTime
 		err = r.client.Status().Update(context.TODO(), buildRun)
 		if err != nil {
 			reqLogger.Error(err, "Failed to update the BuildRun status", "BuildRun", buildRun.Name)
+			buildRun = r.updateBuildRunErrorStatus(buildRun, err)
+			err = r.client.Status().Update(context.TODO(), buildRun)
+			if err != nil {
+				reqLogger.Error(err, "Failed to update the BuildRun status", "BuildRun", buildRun.Name)
+				return reconcile.Result{}, err
```

Comment:
You could delete this `return` statement, and keep the one happening in line 153.

Reply:
Hi @otaviof ,

It is not a duplicated one.

If I skip this return, and the the err in`err = r.client.Status().Update(context.TODO(), buildRun)` is nul, 
then the final return code will become: `return reconcile.Result{}, nil`. It is not correct for us.

But of course, I also refine for a better logic:
```
		buildRun = r.updateBuildRunErrorStatus(buildRun, err)
		updateErr := r.client.Status().Update(context.TODO(), buildRun)
		if updateErr != nil {
			return reconcile.Result{}, updateErr
		}
		return reconcile.Result{}, err
```

I also refine for below comments.

Thanks!

Reply:
Got you. Thanks.